shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -622,8 +623,12 @@ void onRequestPermissionsResult(
   void onNewIntent(@NonNull Intent intent) {
     ensureAlive();
     if (flutterEngine != null) {
-      Log.v(TAG, "Forwarding onNewIntent() to FlutterEngine.");
+      Log.v(TAG, "Forwarding onNewIntent() to FlutterEngine and sending pushRoute message.");
       flutterEngine.getActivityControlSurface().onNewIntent(intent);
+      Uri data = intent.getData();
```

Comment:
there's a little bit of inconsistently between here and the getinitialroute

Have stuff in flutteractivity/fragment if you expect subclass developers to actively decide whether they want to keep or override this behavior. 

If the same behavior's in both activity and fragment, just move them to the delegate (and make sure to document what the default behavior is in the javadoc).